nextstrain/cli/runner/conda.py

```diff
@@ -579,6 +580,21 @@ def package_name(spec: str) -> str:
     return PackageSpec.parse(spec).name
 
 
+def latest_package_label_version(channel: str, package: str, label: str) -> str:
+    response = requests.get(f"https://api.anaconda.org/package/{urlquote(channel)}/{urlquote(package)}/files")
```

Comment:
Hmm, this adds another fetch, presumably to preserve the existing `package_distribution()` return value? We're only using two keys from that return value—`download_url` and `version`—and I think we could get those in a single fetch of the channel `current_repodata.json` instead of the Anaconda API. Internally the filtering would look very similar to what `package_distribution()` does already + the sort/filter the new `latest_package_label_version()` does. And it'd be kinda nice not to rely on the Anaconda API since it's another source for failures (and we already rely on the channel repodata to do install anything with Micromamba anyway).

…

Ahh, but scratch that! While our channel is small, we also have to do this for conda-forge, which is not small, and the fetch time (even on a fast connection, with gzip content encoding) is several seconds which outweighs for sure the extra roundtrip for the fetch.

…

However, I'm also just realizing now that there's another bug in our use of `https://api.anaconda.org/releases/…` too (not fixed by this PR, maybe doubled down on): we assume the API response includes one release per subdir. But it can actually include one release per subdir **per build** when there's multiple builds. e.g.

```
$ curl -s https://api.anaconda.org/release/conda-forge/micromamba/1.0.0 | jq '.distributions | map(select(.attrs.subdir == "linux-64") | {download_url, version, attrs})'
[
  {
    "download_url": "//api.anaconda.org/download/conda-forge/micromamba/1.0.0/linux-64/micromamba-1.0.0-0.tar.bz2",
    "version": "1.0.0",
    "attrs": {
      "arch": "x86_64",
      "build": "0",
      "build_number": 0,
      "depends": [],
      "has_prefix": true,
      "license": "BSD-3-Clause AND MIT AND OpenSSL",
      "license_family": "BSD",
      "machine": "x86_64",
      "operatingsystem": "linux",
      "platform": "linux",
      "subdir": "linux-64",
      "target-triplet": "x86_64-any-linux",
      "timestamp": 1667390305411
    }
  },
  {
    "download_url": "//api.anaconda.org/download/conda-forge/micromamba/1.0.0/linux-64/micromamba-1.0.0-1.tar.bz2",
    "version": "1.0.0",
    "attrs": {
      "arch": "x86_64",
      "build": "1",
      "build_number": 1,
      "depends": [],
      "has_prefix": true,
      "license": "BSD-3-Clause AND MIT AND OpenSSL",
      "license_family": "BSD",
      "machine": "x86_64",
      "operatingsystem": "linux",
      "platform": "linux",
      "subdir": "linux-64",
      "target-triplet": "x86_64-any-linux",
      "timestamp": 1667592485783
    }
  }
]
```

So we have to sort properly by build even when using this API. orz

Reply:
Added in https://github.com/nextstrain/cli/pull/280/commits/a37b4d3642aacf9655dd096ad794266b24de4535